src/phoenix/server/api/types/Span.py

```diff
@@ -166,33 +164,39 @@ async def document_evaluations(self, info: Info[Context, None]) -> List[Document
     @strawberry.field(
         description="Retrieval metrics: NDCG@K, Precision@K, Reciprocal Rank, etc.",
     )  # type: ignore
-    def document_retrieval_metrics(
+    async def document_retrieval_metrics(
         self,
+        info: Info[Context, None],
         evaluation_name: Optional[str] = UNSET,
     ) -> List[DocumentRetrievalMetrics]:
         if not self.num_documents:
             return []
-        span_id = SpanID(str(self.context.span_id))
-        all_document_evaluation_names = self.project.get_document_evaluation_names(span_id)
-        if not all_document_evaluation_names:
-            return []
-        if evaluation_name is UNSET:
-            evaluation_names = all_document_evaluation_names
-        elif evaluation_name not in all_document_evaluation_names:
+        stmt = (
+            select(
+                models.DocumentAnnotation.name,
+                models.DocumentAnnotation.score,
+                models.DocumentAnnotation.document_index,
+            )
+            .where(models.DocumentAnnotation.score != None)  # noqa: E711
+            .where(models.DocumentAnnotation.span_rowid == self.span_rowid)
+            .order_by(models.DocumentAnnotation.name)
+        )
+        if evaluation_name:
```

Comment:
```suggestion
        if evaluation_name is not UNSET:
```

Bit of an edge case, but it would be possible right now for someone to create an eval name with the empty string. Maybe we want to eliminate that possibility.

Reply:
oh dang